sdk/core/Azure.Core/src/AccessToken.cs

```diff
@@ -49,12 +66,17 @@ public AccessToken(string accessToken, DateTimeOffset expiresOn, DateTimeOffset?
         /// </summary>
         public DateTimeOffset? RefreshOn { get; }
 
+        /// <summary>
+        /// Identifies the type of access token.
+        /// </summary>
+        public string TokenType { get; }
+
         /// <inheritdoc />
         public override bool Equals(object? obj)
         {
             if (obj is AccessToken accessToken)
             {
-                return accessToken.ExpiresOn == ExpiresOn && accessToken.Token == Token;
+                return accessToken.ExpiresOn == ExpiresOn && accessToken.Token == Token && accessToken.TokenType == TokenType;
```

Comment:
Do we ever serialize these, and would changing the definition of a property not present in an earlier version be a breaking change?

Reply:
no, they are never serialized - it is used as an output type only.